### PR TITLE
Revise routes.yaml explanations

### PR DESCRIPTION
**Pull request by mrkschan**
- Simplified descriptions
- Mentions generation of HTTPS routes for HTTP route
- Mentions generation of HTTP 301 redirect for HTTPS route
- Unified application name in examples

### CODE REVIEW

File: reference/routes-yaml.md [hunk 1]
```diff
@@ -1,116 +1,123 @@
...
-will get: `http://www---refactorcss-mswy7hzcuhcjw.us.platform.sh/` and
+The yaml file is composed of a list of routes and their configuration.
+A route can either be an absolute URL or a URL template that looks like:
+`http://www.{default}/` or `https://{default}/blog` where `{default}`
+will be substituted by the default fully qualified domain name configured
+in the project. So if your default domain is `example.com`, these
+routes will be resolved to `http://www.example.com/` and
+`https://example.com/blog` in the master environment.
+
+Platform.sh will also generate domain for every active environment.
```
Comment (pjcdawkins) [inline reference/routes-yaml.md]: a domain

File: reference/routes-yaml.md [hunk 2]
```diff
@@ -1,116 +1,123 @@
...
+In those cases, `{default}` will be replaced by
+`[branch]-[project-id].[region].platform.sh`. For example, a
+project with id `mswy7hzcuhcjw` on a Git branch called `refactorcss` hosted
```
Comment (pjcdawkins) [inline reference/routes-yaml.md]: for a project with ID

File: reference/routes-yaml.md [hunk 3]
```diff
@@ -131,55 +139,61 @@ https://{default}/:
...
 to route the url `http://add-theme-vmwklxcpbi6zq.eu.platform.sh/` to this
 environment. If, for example, we also defined a `http://www.{default}/` route,
 you could visit the following url to see
 `http://www---add-theme-vmwklxcpbi6zq.eu.platform.sh/` the same environment.
 
-> **note** notice the triple dash (`---`) we use as a separator for the subdomain.
-> This is what replaces the dot (`.`).
+> **note**
+> Triple dash (`---`) is used as a separator for the subdomain in development
+> environment. It replaces the dot (`.`).
```
Comment (pjcdawkins) [inline reference/routes-yaml.md]: environments

**Comment by mrkschan** [comment]
Revised. Please review.

**Comment by mrkschan** [comment]
Check this out :) https://pr-349-t2llqeifuhpzg.eu.platform.sh/user_guide/reference/routes-yaml.html

**Comment by mrkschan** [comment]
WIP again :) Let's not merge this ...

**Comment by mrkschan** [comment]
No longer provide deterministic domain for dev environments.

File: reference/routes-yaml.md [hunk 4]
```diff
@@ -1,186 +1,168 @@
...
+A route can either be an absolute URL or a URL template that looks like:
+`http://www.{default}/` or `https://{default}/blog` where `{default}`
+will be substituted by the default fully qualified domain name configured
+in the project. So if your default domain is `example.com`, these
+routes will be resolved to `http://www.example.com/` and
+`https://example.com/blog` in the master environment.
+
+Platform.sh will also generate a domain for every active development environment.
+For example, a project with ID `mswy7hzcuhcjw` on a Git branch called
+`refactorcss` hosted in the `us` region, we will get:
```
Comment (pjcdawkins) [inline reference/routes-yaml.md]: `For example, a project ... we will get` is wrong..

Comment (mrkschan) [inline reference/routes-yaml.md]: Do you mean we need the short "random" string in the generated URL as well?

**Pull request merged**